From c843620e6624e4ecdbb7432177025963e96d346a Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Mon, 18 Aug 2014 19:58:54 -0400
Subject: [block] nvme: Use Log Page constants in SCSI emulation

Message-id: <1408391935-24886-22-git-send-email-dmilburn@redhat.com>
Patchwork-id: 87899
O-Subject: [RHEL7.1 PATCH BZ 1111259 21/22] NVMe: Use Log Page constants in SCSI emulation
Bugzilla: 1111259
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Matthew Wilcox <matthew.r.wilcox@intel.com>

The nvme-scsi file defined its own Log Page constant.  Use the
newly-defined one from the header file instead.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
(cherry picked from commit ef351b97dedaa7a6e257ed4f554718e384d8786b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/nvme-scsi.c b/drivers/block/nvme-scsi.c
index 4fc25b9..24308ae 100644
--- a/drivers/block/nvme-scsi.c
+++ b/drivers/block/nvme-scsi.c
@@ -239,7 +239,6 @@ static int sg_version_num = 30534; /* 2 digits for each component */
 #define READ_CAP_16_RESP_SIZE    32
 
 /* NVMe Namespace and Command Defines */
-#define NVME_GET_SMART_LOG_PAGE    0x02
 #define BYTES_TO_DWORDS     4
 #define NVME_MAX_FIRMWARE_SLOT    7
 
@@ -1019,7 +1018,7 @@ static int nvme_trans_log_info_exceptions(struct nvme_ns *ns,
  c.common.nsid = cpu_to_le32(0xFFFFFFFF);
  c.common.prp1 = cpu_to_le64(dma_addr);
  c.common.cdw10[0] = cpu_to_le32((((sizeof(struct nvme_smart_log) /
-   BYTES_TO_DWORDS) - 1) << 16) | NVME_GET_SMART_LOG_PAGE);
+   BYTES_TO_DWORDS) - 1) << 16) | NVME_LOG_SMART);
  res = nvme_submit_admin_cmd(dev, &c, NULL);
  if (res != NVME_SC_SUCCESS) {
   temp_c = LOG_TEMP_UNKNOWN;
@@ -1087,7 +1086,7 @@ static int nvme_trans_log_temperature(struct nvme_ns *ns, struct sg_io_hdr *hdr,
  c.common.nsid = cpu_to_le32(0xFFFFFFFF);
  c.common.prp1 = cpu_to_le64(dma_addr);
  c.common.cdw10[0] = cpu_to_le32((((sizeof(struct nvme_smart_log) /
-   BYTES_TO_DWORDS) - 1) << 16) | NVME_GET_SMART_LOG_PAGE);
+   BYTES_TO_DWORDS) - 1) << 16) | NVME_LOG_SMART);
  res = nvme_submit_admin_cmd(dev, &c, NULL);
  if (res != NVME_SC_SUCCESS) {
   temp_c_cur = LOG_TEMP_UNKNOWN;
-- 
1.7.1